python_modules/dagster-graphql/dagster_graphql/schema/asset_graph.py

```diff
@@ -862,12 +862,20 @@ def resolve_partitionStats(
                 ]
             )
 
+            num_failed_and_not_in_progress = len(
```

Comment:
if we add many more statuses we'll want a smarter way to do this, akin to the flatten method

Reply:
nit: seems like this would call `in_progress_subset.get_partition_keys()` k times? can we pull that out so we only fetch it once